src/libraries/iterableMap.se

```diff
@@ -28,7 +29,14 @@ def addOrUpdate(key, value):
     else:
         self.update(key, value)
 
-# TODO: add support for item removal, note that this requires moving the last item in the collection to the newly opened slot so we don't end up with a sparse collection
+def remove(key):
+    require(self.contains(key))
+    # move last item in collection to the newly opened slot
+    self.itemsArray[self.itemsMap[key].offset] = self.getByOffset(self.numberOfItems)
```

Comment:
What happens if item being removed is the last item?  What happens if item being removed is the only item?  I haven't thought through these, so maybe this code works with them already, but wanted to make sure someone thinks about it.